src/platforms/ios/iosnetworkwatcher.mm

```diff
@@ -0,0 +1,117 @@
+    // the default-route-transport  is not the vpn-tunnel
+    return mCurrentDefaultTransport;
+  }
+  if (mObservableConnection != nil) {
+    return mCurrentVPNTransport;
+  }
+  // If we don't have an open tunnel-observer, mCurrentVPNTransport is probably wrong.
+  return NetworkWatcherImpl::TransportType_Unknown;
+}
+
+NetworkWatcherImpl::TransportType IOSNetworkWatcher::toTransportType(nw_path_t path) {
+  if (path == nil) {
+    return NetworkWatcherImpl::TransportType_Unknown;
+  }
+  auto status = nw_path_get_status(path);
+  if (status != nw_path_status_satisfied && status != nw_path_status_satisfiable) {
+    // We're offline.
+    return NetworkWatcherImpl::TransportType_None;
+  }
+  if (nw_path_uses_interface_type(path, nw_interface_type_wifi)) {
+    return NetworkWatcherImpl::TransportType_WiFi;
+  }
+  if (nw_path_uses_interface_type(path, nw_interface_type_wired)) {
+    return NetworkWatcherImpl::TransportType_Ethernet;
+  }
+  if (nw_path_uses_interface_type(path, nw_interface_type_cellular)) {
+    return NetworkWatcherImpl::TransportType_Cellular;
+  }
+  if (nw_path_uses_interface_type(path, nw_interface_type_other)) {
+    return NetworkWatcherImpl::TransportType_Other;
+  }
+  if (nw_path_uses_interface_type(path, nw_interface_type_loopback)) {
+    return NetworkWatcherImpl::TransportType_Other;
+  }
+  return NetworkWatcherImpl::TransportType_Unknown;
+}
+
+void IOSNetworkWatcher::controllerStateChanged() {
+  if (MozillaVPN::instance()->controller()->state() != Controller::StateOn) {
+    if (mObservableConnection != nil) {
+      nw_connection_cancel(mObservableConnection);
+      nw_release(mObservableConnection);
+      mObservableConnection = nil;
+    }
+    return;
+  }
+  // When connected the current default route is the VPN (ofc)
+  // therefore the monitor will report transport::other.
+  // We need to open up a theoretical socket to our entry-node, to find out
+  // what connection the route is using
+  auto vpn = MozillaVPN::instance();
+  // When multihop is used, we need to connect to the entry server,
+  // otherwise the exit server is the target
+  auto key = vpn->multihop() ? vpn->entryServerPublicKey() : vpn->serverPublicKey();
+  auto serverlist = vpn->multihop() ? vpn->entryServers() : vpn->exitServers();
+  auto index = serverlist.indexOf(key);
+  // No such server
+  if (index == -1) {
+    return;
+  }
+  auto server = serverlist.at(index);
+  auto str = server.ipv4AddrIn().toStdString();
+  auto endpoint = nw_endpoint_create_host(str.c_str(), "80");
+  auto params = nw_parameters_create_secure_udp(NW_PARAMETERS_DISABLE_PROTOCOL,
+                                                NW_PARAMETERS_DEFAULT_CONFIGURATION);
+  auto con = nw_connection_create(endpoint, params);
+  nw_connection_set_queue(con, sQueue);
+  nw_connection_set_path_changed_handler(con, ^(nw_path_t _Nonnull path) {
+    mCurrentVPNTransport = toTransportType(path);
+  });
+  nw_connection_start(con);
```

Comment:
in this way you create a connection, but how to keep it alive? Do we need to?

Reply:
I haven't found anything in the docs for that. But as it's a udp connection i would suspect it's "stateless" and therefore not really getting closed by the os for not using it.